### PR TITLE
style-components:^4.1.3 & dev -> devDeps || peerDeps

### LINKED ISSUE #402: Organise peerDependencies for all components
**Is your feature request related to a problem? Please describe.**
We should evaluate the `dependencies` of each of our components to ensure that we utilise both `peerDependencies` and `devDependencies` correctly. 

`peerDependencies` are required when the component needs a dependency for production but we do not want duplicates of the dependency in the component and the application which uses the component. EG: `react` and `styled-components`

`devDependencies` are required when the component needs a dependency for development, such as unit tests or stories. An example of this would be a unit test which created a React component to inject into the component being tested. EG: `React` or `styled-components`

`peerDependencies` and `devDependencies` are not mutually exclusive. 

Currently, if we `npm link` components from psammead to simorgh we get a duplicate of `styled-components` meaning the visual render is buggy. We also have `react` as a duplicate dependency in some places which is mitigated via treeshaking but we should fix this in the packages to ensure other applications without tree shaking won't have `react` on the page multiple times.

**Describe the solution you'd like**
Evaluate the dependencies of each package. Especially moving `react` and `styled-components`.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Unit and cypress tests  won't change, a visual regression of each component is necessary. 

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #402 

**Overall change:** _Moves styled-components and react out of `dependencies` and into `devDependencies` && `peerDependencies`_

**Code changes:**

-  _Moves styled-components and react out of `dependencies` and into `devDependencies` && `peerDependencies`_

I suggest that Test perform a basic regression of each psammead component. This PR purely changes the dependency structure therefore there should be no regression with visual, behavioural or unit tests for each component

---

- [x] I have assigned myself to this PR and the corresponding issues
- ~[ ] Tests added for new features~
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-figure/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
-| 0.1.10   | [PR#392](https://github.com/BBC/psammead/pull/392) Increases the padding-bottom from 16px to 24px |
+| 0.1.11  | [PR#407](https://github.com/bbc/psammead/pull/407) Organise dependencies properly |
+| 0.1.10  | [PR#392](https://github.com/BBC/psammead/pull/392) Increases the padding-bottom from 16px to 24px |
```
Comment (pjlee11) [inline packages/components/psammead-figure/CHANGELOG.md:7]: Single whitespace change `|-0.1.10---|` to become `|-0.1.10--|` so the table lines up better

File: packages/components/psammead-visually-hidden-text/CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 | Version | Description |
 | --------------------- |
-| 0.3.10  | [PR#323](https://github.com/bbc/psammead/pull/323) Update storybook badge url |
```
Comment (pjlee11) [inline packages/components/psammead-visually-hidden-text/CHANGELOG.md:5]: This was a [version typo](https://github.com/bbc/psammead/pull/323/files#diff-5d2b9269702cadef473c4286f20d06ccR5)

**Comment by pjlee11** [comment]
Question for reviewers - Should we be explicitly specifying `react` as a `peerDependency` for all components? 

Some pass unit tests and render in storybook fine without `react` listed in the deps but would always need `react` to render in an application.

**Comment by dr3** [comment]
@pjlee11 if a component uses react directly, it should be a peer. If its only used for tests, its a dev, if its not used it shouldn't be one

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
   "dependencies": {
     "@bbc/psammead-styles": "^0.3.0",
     "@storybook/theming": "^5.0.1",
-    "styled-components": "^4.1.2",
+    "styled-components": "^4.1.3",
```
Comment (dr3) [inline package.json:38]: This requires the base level psammead to be bumped and changeloged

File: packages/components/psammead-image/package.json [hunk 4]
```diff
@@ -16,14 +16,16 @@
...
   "devDependencies": {
     "@bbc/psammead-assets": "^0.1.3",
     "@bbc/psammead-test-helpers": "^0.3.0",
-    "react": "^16.6.3"
+    "react": "^16.6.3",
+    "prop-types": "^15.6.2",
+    "styled-components": "^4.1.3"
+  },
+  "peerDependencies": {
+    "prop-types": "^15.6.2",
```
Comment (dr3) [inline packages/components/psammead-image/package.json:27]: Hmmmmmmmmmmmm idk about `prop-types` being a peer, ile allow it :P

**Review by dr3 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (pjlee11) [reply package.json:38]: done

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 1.1.2   | [PR#407](https://github.com/BBC/psammead/pull/407) Change styled-component version to be consistent everywhere |
```
Comment (dr3) [inline CHANGELOG.md]:
```suggestion
| 1.1.2   | [PR#407](https://github.com/BBC/psammead/pull/407) Change styled-component version to be consistent everywhere and use peer dependancies |
```

**Review by dr3 — DISMISSED** [review]
Yassssss

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 1.1.2   | [PR#407](https://github.com/BBC/psammead/pull/407) Change styled-component version to be consistent everywhere and use peer dependancies |
```
Comment (ChrisBAshton) [inline CHANGELOG.md]:
```suggestion
| 1.1.2   | [PR#407](https://github.com/BBC/psammead/pull/407) Change styled-component version to be consistent everywhere and use peer dependencies |
```

File: packages/components/psammead-copyright/package.json [hunk 7]
```diff
@@ -20,15 +20,18 @@
     "@bbc/gel-foundations": "^0.3.0",
     "@bbc/psammead-styles": "^0.3.2"
   },
+  "devDependencies": {
+    "@bbc/psammead-test-helpers": "^0.3.3",
+    "@bbc/psammead-visually-hidden-text": "^0.1.8"
```
Comment (ChrisBAshton) [inline packages/components/psammead-copyright/package.json]: Looks like we're missing `styled-components` from our `devDependencies` here.

File: packages/components/psammead-image/package.json [hunk 8]
```diff
@@ -16,14 +16,16 @@
...
+    "styled-components": "^4.1.3"
```
Comment (ChrisBAshton) [inline packages/components/psammead-image/package.json:29]: Shouldn't react be a peerDependency as well as a devDependency? Like you've done here: https://github.com/bbc/psammead/pull/407/files#diff-c11da1456f9036b64ebb3664660409a2R30

File: packages/components/psammead-image-placeholder/package.json [hunk 9]
```diff
@@ -18,12 +18,15 @@
...
+    "@bbc/psammead-styles": "^0.1.3"
   },
   "devDependencies": {
     "@bbc/psammead-test-helpers": "^0.3.0",
-    "react": "^16.6.3"
+    "react": "^16.6.3",
+    "styled-components": "^4.1.3"
+  },
+  "peerDependencies": {
+    "styled-components": "^4.1.3"
```
Comment (ChrisBAshton) [inline packages/components/psammead-image-placeholder/package.json:30]: Shouldn't react be a peerDependency too? As it's used in index.jsx

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]
Have made the assumption that:

- react should be a peerDependency of any component that uses React directly (as opposed to via styled-components) on the front-end
- styled-components, whilst literally dependencies of all of our Psammead components, cannot be listed as `dependencies` because otherwise upstream applications may have a conflicting version of `styled-components`, meaning more than 1 version may exist in node_modules, and as it's a singleton we may be missing some styles.

^ NB, if that's the case, shouldn't we make `styled-components` dependencies of each component and then add clear instructions to upstream applications that they must specify `styled-components` in their `peerDependencies`?

File: packages/components/psammead-image-placeholder/package.json (hunk 9 above)
Comment (pjlee11) [reply packages/components/psammead-image-placeholder/package.json:30]: yep

File: packages/components/psammead-image/package.json (hunk 8 above)
Comment (pjlee11) [reply packages/components/psammead-image/package.json:29]: yep

File: packages/components/psammead-copyright/package.json (hunk 7 above)
Comment (pjlee11) [reply packages/components/psammead-copyright/package.json]: good spot

**Review by dr3 — APPROVED** [review]

**Review by ChrisBAshton — APPROVED** [review]
Spoke IRL and it makes sense - `styled-components` in `devDependencies` where its used by Storybook / Unit tests. `styled-components` in `peerDependencies` where the component says "I need this, but I won't install it myself" and it's up to the parent to install it.

Happy to approve on this basis :+1: thanks Phil

**Comment by pjlee11** [comment]
[Peer dependencies explained](https://nodejs.org/en/blog/npm/peer-dependencies/)

**Pull request merged**